parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveType.java

```diff
@@ -86,6 +86,14 @@ public void addValueToPrimitiveConverter(
       public <T, E extends Exception> T convert(PrimitiveTypeNameConverter<T, E> converter) throws E {
         return converter.convertINT64(this);
       }
+
+      @Override
+      PrimitiveComparator<?> comparator(OriginalType logicalType) {
+        if (logicalType == OriginalType.UINT_64)
+          // TODO: return unsigned comparator
+          return PrimitiveComparator.SIGNED_INT64_COMPARATOR;
```

Comment:
I know this is only temporary, but it's probably a better practice to throw `UnsupportedOperationException` rather than return the wrong comparator. That way if it is overlooked later, it fails instead of silently introducing correctness problems.

Reply:
My concept here is to refactor the code to the new design with still passing unit tests. Then, at the final step add the new comparators and fix the existing tests and add more to pass.

Reply:
Makes sense.